Restructure ORIENTATION.md (#51)
the-collab-lab/smart-shopping-list: ORIENTATION.md

# The Collab Lab Orientation

## What is The Collab Lab?

> The Collab Lab exists to help early-career engineers gain practical experience by working remotely on real world projects with other early-career developers.

Reviewer: I suggest swapping out "practical experience" with something more like "experience working collaboratively on a software team"

> The Collab Lab exists to help early-career engineers gain practical experience by working remotely on real world projects with other early-career developers.

Reviewer: ```suggestion
The Collab Lab exists to help early-career engineers gain experience working collaboratively on a software team by working remotely on real world projects with other early-career developers.
```

The goal of this project is to work together to build a “smart” shopping list that learns your shopping habits over time, putting items you are most likely to need to buy in the near future at the top.

> Every team builds their app from the same set of instructions, but each comes out a little different and reflecting the personality of the group.

Reviewer: change

> different and reflecting

to

> different, reflecting

(this is a suggestion, definitely not hill-dying status!)

> Every team builds their app from the same set of instructions, but each comes out a little different and reflecting the personality of the group.

Reviewer: ```suggestion
Every team builds their app from the same set of instructions, but each comes out a little different, reflecting the personality of the group.
```

> :tv: &nbsp; Check out a video demo of the app team #6 made here: 

Reviewer: does `:tv:` work here?

> [![TCL Demo Video Screenshot](_resources/images/screenshots/TCLDemoReadmePic.png)](https://youtu.be/uWgkwY_VBPo)

Reviewer: whoa, that's some fancy, nested markdown! don't think I've ever seen that before

> [![TCL Demo Video Screenshot](_resources/images/screenshots/TCLDemoReadmePic.png)](https://youtu.be/uWgkwY_VBPo)

Author: ```suggestion
[![TCL Demo Video Screenshot](_resources/images/TCLDemoReadmePic.png)](https://youtu.be/uWgkwY_VBPo)
```

## Who am I working with?  

When new developers and mentors join the Collab Lab community, they join a group of brilliant and supportive friends that stretch across the globe. Every quarter, a new cohort starts that consists of several regionally-based teams. Those teams are comprised of four developers who collaborate to complete an app of their own and three mentors to help support them through the project. 

## Why do we do what we do? 

> Knowing how to code is just one aspect of being a professional web developer. Another super important skill is knowing how to collaborate effectively on a software team -- that’s where Collab Lab comes into play! 

Reviewer: change

> team -- that’s

to 

> team—that’s

> Knowing how to code is just one aspect of being a professional web developer. Another super important skill is knowing how to collaborate effectively on a software team -- that’s where Collab Lab comes into play! 

Reviewer: ```suggestion
Knowing how to code is just one aspect of being a professional web developer. Another super important skill is knowing how to collaborate effectively on a software team—that’s where Collab Lab comes into play!
```

### Pair Programming:

> A portion of your dedicated hours each week will be spent [pair programming](https://www.freecodecamp.org/news/how-remote-pair-programming-works-and-why-it-can-change-your-life-cd7b767dc60f/) with your partner — this is where the most magical collaboration happens.

Reviewer: ```suggestion
A portion of your dedicated hours each week will be spent [pair programming](https://www.freecodecamp.org/news/how-remote-pair-programming-works-and-why-it-can-change-your-life-cd7b767dc60f/) with your partner—this is where the most magical collaboration happens.
```

> A portion of your dedicated hours each week will be spent [pair programming](https://www.freecodecamp.org/news/how-remote-pair-programming-works-and-why-it-can-change-your-life-cd7b767dc60f/) with your partner—this is where the most magical collaboration happens.

Author: ```suggestion
A portion of your dedicated hours each week will be spent [pair programming](https://www.microverse.org/blog/how-remote-pair-programming-works-and-why-it-can-change-your-life) with your partner—this is where the most magical collaboration happens.
```

> Pair programming offers some really exciting opportunities to teach/learn from your partner, get a different perspective and/or insight on the issues you’re facing, and drive forward progress on your goals.

Reviewer: nice, love this!!!

> There is no one “right” way to pair program, but in general, you will meet with your partner on Zoom to work through the story or task you’ve chosen for that week. You’re encouraged to test out some different approaches to pair programming to see what works best for your and your partner’s collaboration styles.

Author: ```suggestion
There is no one “right” way to pair program, but in general you will meet with your partner on Zoom to work through the story or task you’ve chosen for that week. You’re encouraged to test out different approaches to pair programming to see what works best for your and your partner’s collaboration styles.
```

## How does it all work?  

> The project is organized in GitHub as a set of [user stories](https://www.mountaingoatsoftware.com/agile/user-stories), each with a description of the desired functionality as well as [acceptance criteria](https://www.leadingagile.com/2014/09/acceptance-criteria/) (AC) that describe how you know whether the task or story is complete. You can find the stories on [the project board]({link to the cohort repo project board}) on GitHub. This is referred to as  the “backlog” (the collection of stories) and represents the work needed to complete the project. 

Author: ```suggestion
The project is organized in GitHub as a set of [user stories](https://www.mountaingoatsoftware.com/agile/user-stories), each with a description of the desired functionality as well as [acceptance criteria](https://www.leadingagile.com/2014/09/acceptance-criteria/ (AC)) that describe how you know whether the task or story is complete. You can find the stories on [the project board]({link to the cohort repo project board}) on GitHub. This is referred to as the “backlog” (the collection of stories) and represents the work needed to complete the project. 
```

- The code to complete the task or implement the feature has been reviewed and approved by the other pair of developers
- The [Product Owner](https://www.agilealliance.org/glossary/product-owner/) (i.e. one of the mentors) has accepted the work as satisfactorily meeting the requirements

> Each week, the team will split into pairs and each pair will complete a single story from the backlog. 

Reviewer: ```suggestion
Each week, the team will split into pairs and each pair will be responsible for completing a single story from the backlog.
```

At the end of the 8 week project all of the stories will be complete and we will have a beautifully functioning app that the team has collaboratively built! 

### Coordination & communication

It’s super helpful both to yourself and the mentors for you to document your work. This can prevent you from losing track of details or decisions you make with your pair buddy. It also allows mentors to catch up on your progress when you ask for help.

One of the first things you & your pair buddy should do each week is create a [draft PR](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests#draft-pull-requests) for your work. This allows you to have a conversation about the work right alongside the code. Coupled with questions and discussions about your approach in the issue itself, this is nice way of documenting your progress as you go.

### Slack

Most of our team coordination will happen through [the Collab Lab Slack team](https://the-collab-lab.slack.com) and in particular your team channel.

If you’re not already super familiar with Slack, no worries! Here are a couple of tips to help you use it like a pro!

> 1. Threads threads threads! No, seriously, threads! 😅 Threads are great because they do 2 really helpful things: 1) They encapsulate conversations so future you is able to find both the question and the answer in one place, 2) They keep notifications to a minimum, giving people the option of following a conversation or not.

Author: ```suggestion
1. Threads threads threads! No seriously, threads! 😅  Threads are great because they do 2 really helpful things:
    * They encapsulate conversations so the future you is able to find both the question and the answer in one place
    * They keep notifications to a minimum, giving people the option of following a conversation or not.
```

2. Speaking of notifications, tag people in when you need their attention. “`@stacie`, could you review this PR?” is better than “Can someone review this PR?” because the latter may not be seen for a while depending on the person’s notification settings. When you need to alert the whole channel to something, use `@here`. We also have groups set up such as `@group-tcl-9` and `@group-tcl-9-mentors` as shortcuts to tagging in certain groups. Use them!
3. Use GIFs and emoji liberally! Communicating via text can come across really dry. Don’t be afraid to communicate a little emotion along with that question about React hooks! 🤪

> **Note:** You can disable the animation of gifs and emojus in the Slack app itself. Visit [the Slack Help Center](https://slack.com/help/articles/228023907-Manage-animated-images-and-emoji) to learn more.

Reviewer 2: @segdeha Done 👍

- Saturday — Respond to mentor feedback on PRs
- Sunday — Demo on production!

Having watched many cohorts before yours, our advice is to front-load the work as much as possible in the first couple of days of each week. Some of the stories are trickier than they appear on the surface!

### Weekly Syncs 

Every Sunday, the team will gather for 1-hour on Zoom for a Weekly Sync. Those meetings will always follow a familiar pattern: 

- Demos/discussion (15 minutes)
  - The call will always start with each pair presenting a quick demo of the feature they built in “production” (the live version of the app that is deployed to the web for the world to see) followed by a concise walk-through of the code they wrote to build it. This is a common practice on most software teams and also lets you practice talking about your work, which can come in handy in a job interview setting!
- Learning module (30 minutes every other week)
  - Presentations by mentors on topics that align with what your team is learning. Examples: git, code reviewing, pair programming, communication for developers. 
- Retrospective (30 minutes every other week)

>   - In our retros, we will share appreciations of our teammates as well as talk about what went well from a process perspective and what we could do to improve how we’re working together.

Reviewer: I like this...except I'd keep that first comma 😬 

Val says it's a matter of preference, so the important thing it to be consistent across the document. I've lost track at this point whether we're keeping or dropping commas in this case 😛
